File: cp_nim/src/bin/abc355-b-piano-2.nim
# Created by Ayush Biswas at 2026/07/19 11:24
# https://atcoder.jp/contests/adt_easy_20240702_1/tasks/abc355_b
import ../lib/cpio

# @code begin

import sugar, sequtils, algorithm

var r = newReader()

var
  (n, m) = r[int, int]
  a = r[seq[int]].map(x => (x, 0))
  b = r[seq[int]].map(x => (x, 1))

a.add b
a.sort

if (1 .. a.high).anyIt(a[it - 1][1] == 0 and a[it][1] == 0):
  echo "Yes"
else:
  echo "No"

# @code end

File: cp_rust/src/bin/abc355-b-piano-2.rs
// Created by Ayush Biswas at 2026/07/11 21:44
// https://atcoder.jp/contests/adt_easy_20240702_1/tasks/abc355_b
use cp_lib::*;

// @code begin
use cpio::*;
use itertools::Itertools;

sol! {
    fn solution(
        [_n, _m]: [usize; 2],
        a: [usize],
        b: [usize]
    ) -> Bool {
        let a_tagged = a.into_iter().map(|ai| (ai, 1));
        let b_tagged = b.into_iter().map(|bi| (bi, 0));
        a_tagged
            .chain(b_tagged)
            .sorted()
            .map(|(_, tag)| tag)
            .scan(0, |acc, ai| {
                let res = *acc * ai;
                *acc = ai;
                Some(res)
            })
            .any(|m| m == 1).into()
    }
}
// @code end
